jwt/src/main/java/me/uport/sdk/jwt/JWTTools.kt

```diff
@@ -237,7 +226,7 @@ class JWTTools(
 
         val signatureIsValid = verificationMethod[header.alg]
                 ?.invoke(publicKeys, sigData, signingInputBytes)
-                ?: throw InvalidAlgorithmParameterException("JWT algorithm ${header.alg} not supported")
+                ?: throw JWTEncodingException("JWT algorithm ${header.alg} not supported")
```

Comment:
The `JWTEncodingException` change is a good move, then we can have more specific details in the message.

It may be worth us reviewing other exceptions to follow a similar pattern

Reply:
There are still some places where "TooGenericException" is thrown or caught.
Next detekt cleanup should be the place to iron out all of them